packages/@storefront/sayt/src/previous-search-terms/index.ts

```diff
@@ -0,0 +1,37 @@
+import { provide, tag, Events, Selectors, Store, Tag } from '@storefront/core';
+import Sayt from '../sayt';
+
+@provide('previousSearch', (props) => props)
+@tag('gb-sayt-previous-search', require('./index.html'))
+class PreviousSearch {
+  props: PreviousSearch.Props = {
+    onClick: (query) => () => this.actions.search(query),
+  } as any;
+
+  constructor() {
+    const previousSearches = []
+    this.state = {...this.state, previousSearches}
```

Comment:
Since we're using `previousSearches` immediately, lets inline the assignment like so:

```
this.state = { previousSearches: [] };
```

Also, since this component doesn't extend an existing class, we don't need to spread `this.state`.

Reply:
Also, since this is TypeScript, you can simply define this outside like you do with `props`. No need for a `constructor()` function.